config/config-server/src/com/thoughtworks/go/service/ConfigRepository.java

```diff
@@ -390,8 +390,23 @@ void cleanAndResetToMaster() throws IOException {
             deleteBranch(BRANCH_AT_HEAD);
         } catch (Exception e) {
             String currentBranch = git.getRepository().getBranch();
-            LOGGER.error(format("Error while trying to clean up config repository, CurrentBranch: %s \n : \n Message: %s", currentBranch, e.getMessage(), e.getStackTrace()), e);
+            LOGGER.error("Error while trying to clean up config repository, CurrentBranch: {} \n : \n Message: {} \n StackTrace: {}", currentBranch, e.getMessage(), e.getStackTrace(), e);
             throw new RuntimeException(e);
         }
     }
+
+    public void garbageCollect() throws Exception {
+        doLocked(new VoidThrowingFn<Exception>() {
+            public void run() throws Exception {
+                try {
+                    LOGGER.info("Before GC: {}", git.gc().getStatistics());
+                    git.gc().setAggressive(systemEnvironment.get(SystemEnvironment.GO_CONFIG_REPO_GC_AGGRESSIVE)).call();
+                    LOGGER.info("After GC: {}", git.gc().getStatistics());
```

Comment:
The log file entry would look something like this:
```
2015-11-05 19:31:00,029  INFO [pool-1-thread-1] ConfigRepository:403 - Before running GC: {sizeOfPackedObjects=0, sizeOfLooseObjects=6640456, numberOfPackedObjects=0, numberOfPackFiles=0, numberOfPackedRefs=0, numberOfLooseRefs=2, numberOfLooseObjects=2037}
2015-11-05 19:31:02,199  INFO [pool-1-thread-1] ConfigRepository:405 - After running GC: {sizeOfPackedObjects=415262, sizeOfLooseObjects=0, numberOfPackedObjects=2037, numberOfPackFiles=2, numberOfPackedRefs=1, numberOfLooseRefs=1, numberOfLooseObjects=0}
```
That calls out the class name `ConfigRepository`, so I guess that should be ok.